view_all_set.php

```diff
@@ -106,31 +106,20 @@
 # 	26: $f_show_profile
 
 # Set new filter values.  These are stored in a cookie
-$t_view_all_cookie_id = gpc_get_cookie( config_get( 'view_all_cookie' ), '' );
-$t_view_all_cookie = filter_db_get_filter( $t_view_all_cookie_id );
-
-# process the cookie if it exists, it may be blank in a new install
-if( !is_blank( $t_view_all_cookie ) ) {
-	$t_setting_arr = filter_deserialize( $t_view_all_cookie );
-	if( false === $t_setting_arr ) {
-		# couldn't deserialize, if we were trying to use the filter, clear it and reload
-		# for ftype = 0, 1, or 3, we are going to re-write the filter anyways
-		if( !in_array( $f_type, array( 0, 1, 3 ) ) ) {
-			gpc_clear_cookie( 'view_all_cookie' );
-			error_proceed_url( 'view_all_set.php?type=0' );
-			trigger_error( ERROR_FILTER_TOO_OLD, ERROR );
-			exit; # stop here
-		}
-	} else {
-		$t_setting_arr = filter_ensure_valid_filter( $t_setting_arr );
-	}
-} else {
+$t_cookie_name = config_get( 'view_all_cookie' );
+$t_cookie_filter_id = gpc_get_cookie( $t_cookie_name, null );
+if( null === $t_cookie_filter_id ) {
 	# no cookie found, set it
 	$f_type = 1;
+} else {
+	$t_setting_arr = filter_get( $t_cookie_filter_id, filter_get_default() );
 }
 
 # Clear the source query id.  Since we have entered new filter criteria.
-$t_setting_arr['_source_query_id'] = '';
+if( isset( $t_setting_arr['_source_query_id'] ) ) {
```

Comment:
Should we create a constant for strings like `_source_query_id`?

Reply:
Filter properties starting with "_" should be replaced with constant names. This have been discussed for some time.
The inconvenience is that this will touch a lot of code, and i have some parallel work in progress for more incremental changes to filters. That would cause some extensive conflicts for no functionality gains.
I have it in the to-do list, when there is some stabilization between PRs

Reply:
ok